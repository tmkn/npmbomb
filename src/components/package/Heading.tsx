/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import React, { useContext } from "react";

import { AppContext } from "../../App";
import { mq, serifFont, primaryColor } from "../../css";
import { getNameVersion } from "./Package";

interface IHeadingProps {
    packageName: string;
    scope: string | undefined;
}

export const Heading: React.FC<IHeadingProps> = ({ packageName, scope }) => {
    const {
        appState: { inGameMode: gameMode, guesses, remaining }
    } = useContext(AppContext);
    const headerStyle = css({
        [mq[0]]: {
            display: "flex",
            flexWrap: "wrap",
            alignItems: "center",
            paddingBottom: "1rem",
            "& h1": {
                flex: 1
            },
            "& span": {
                fontFamily: serifFont,
                color: primaryColor,
                marginRight: "1rem"
            },
            "& div": {
                display: "flex"
            }
        }
    });

    const current: number = guesses.length + 1;
    const all: number = guesses.length + remaining.length;
    const title: string = scope ? `${scope}/${packageName}` : packageName;
    const downloadUrl: string = scope
        ? `${scope}/${getNameVersion(packageName)[0]}`
        : getNameVersion(packageName)[0];

    return (
        <div css={headerStyle}>
            <h1>{title}</h1>
            {gameMode && (
                <span>
                    [{current}/{all}]
                </span>
            )}
            <div>
                <img src={`https://img.shields.io/npm/dw/${downloadUrl}?style=for-the-badge`} />
            </div>
        </div>
    );
};
