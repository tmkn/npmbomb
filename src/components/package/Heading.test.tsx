import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import { Heading } from "./Heading";

describe("<Heading />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(<Heading name={"test"} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
