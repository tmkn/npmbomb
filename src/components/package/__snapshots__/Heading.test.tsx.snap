// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-44f025"
  >
    <h1>
      test
    </h1>
  </div>
</DocumentFragment>
`;
